docs/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,26 @@
+# Troubleshooting
+
+## Known Problems + Solutions
+
+1) If you uninstall Karpenter after creating some nodes with Karpenter, you will be left with nodes that have finalizers that need to be patched out for the node object to be deleted.
+
+To do this, you can do either of the following:
+- `kubectl edit node <node_name>` and remove the line that says `karpenter.sh/termination` in the finalizers field.
+- Run the following script that gets all nodes with the finalizer and removes all the finalizers from those nodes. NOTE: this will remove ALL finalizers from nodes with the karpenter finalizer.
+```{bash}
+kubectl get nodes -ojsonpath='{range .items[*].metadata}{@.name}:{@.finalizers}{"\n"}' | grep "karpenter.sh/termination" | cut -d ':' -f 1 | xargs kubectl patch node --type='json' -p='[{"op": "remove", "path": "/metadata/finalizers"}]'
```

Comment:
Can you use the in operator to avoid an extra git call?
https://stackoverflow.com/questions/30030652/filter-the-json-according-to-string-in-an-array-in-jsonpath

Reply:
Looks like https://kubernetes.io/docs/reference/kubectl/jsonpath/ recommends JQ if regex is necessary (though maybe direct match works?)

Reply:
It seems like we're switching to use more jsonpath than jq in our docs, and I'd like to keep it similar. Just based off the links you posted, it seems like there are different opinions on whether to use regex in jq or jsonpath. I opted for grep here as we won't take any actions on nodes with finalizers that don't contain `karpenter.sh/termination`.